app.py

```diff
@@ -0,0 +1,73 @@
+import logging
+import os
+from typing import Optional
+
+from flask import Flask, send_from_directory
+from flask.helpers import locked_cached_property
+from jinja2 import ChoiceLoader, FileSystemLoader
+from werkzeug.exceptions import NotFound
+
+import superset
+from superset.initialization import SupersetAppInitializer
+
+logger = logging.getLogger(__name__)
+
+
+# from superset source code
+def create_app(superset_config_module: Optional[str] = None) -> Flask:
+    app = CustomSupersetApp(__name__)
+
+    try:
+        # Allow user to override our config completely
+        config_module = superset_config_module or os.environ.get(
+            "SUPERSET_CONFIG", "superset.config"
+        )
+        app.config.from_object(config_module)
+
+        app_initializer = app.config.get("APP_INITIALIZER", SupersetAppInitializer)(app)
+        app_initializer.init_app()
+
+        return app
+
+    # Make sure that bootstrap errors ALWAYS get logged
+    except Exception as ex:
+        logger.exception("Failed to create app")
+        raise ex
+
+
+class CustomSupersetApp(Flask):
+
+    # superset package paths
+    template_paths = [f"{superset.__path__[0]}/templates"]
+    static_paths = [f"{superset.__path__[0]}/static"]
+
+    # base fn extract from Flask's source code
+    def send_static_file(self, filename):
+        """Iterate over local static folder and superset's one when serving static file"""
+        if not self.has_static_folder or not self.static_folder:
+            raise RuntimeError("'static_folder' must be set to serve static_files.")
+
+        # send_file only knows to call get_send_file_max_age on the app,
+        # call it here so it works for blueprints too.
+        max_age = self.get_send_file_max_age(filename)
+
+        candidates = self.static_paths[:]
```

Comment:
Quelle est l'utilité de `[:]` ?

Reply:
Fait une copie. J'imagine qu'il faut éviter de muter `self.static_paths` (vu dans un snippet).

Reply:
Hmm, I used to know that...